tests/unit/mathematics/linalg/SGLinalg_unittest.cc

```diff
@@ -0,0 +1,72 @@
+#include <shogun/lib/config.h>
+
+#include <shogun/mathematics/linalg/LinalgNamespace.h>
+
+#ifdef HAVE_VIENNACL
+#include <shogun/mathematics/linalg/LinalgBackendViennaCL.h>
+#endif
+
+#include <shogun/lib/SGVector.h>
+#include <gtest/gtest.h>
+
+using namespace shogun;
+
+TEST(LinalgRefactor, linalg_cpu_backend_dot)
+{
+	const index_t size = 10;
+	SGVector<int32_t> a(size), b(size);
+	a.set_const(1);
+	b.set_const(2);
+
+	auto result = linalgns::dot(a, b);
+
+	EXPECT_NEAR(result, 20, 1E-15);
+}
+
+TEST(LinalgRefactor, linalg_gpu_backend_dot_without_gpu_backend)
+{
+	const index_t size = 10;
+	SGVector<int32_t> a(size), b(size), a_gpu, b_gpu;
+	a.set_const(1);
+	b.set_const(2);
+
+	a_gpu.set(linalgns::to_gpu(a));
```

Comment:
this is not how we designed the API in the file I sent you. There we created a vector and then called "to_gpu" on that vector, returning a new vector. And then this vector is passed to linalg::foo